### PR TITLE
python312Packages.json2html: init at 1.3.0

### PR DESCRIPTION
**Pull request by Tochiaha**
Release: https://github.com/softvar/json2html/releases/tag/v1.3.0

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/json2html/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchFromGitHub {
+    owner = "softvar";
+    repo = "json2html";
+    rev = "v${version}";
+    hash = "sha256-r8kAQc6Lc1l2WfUNYnazkdpRLcI5Oz/xqK/I9I7DzEk=";
+  };
+
+  build-systems = [ setuptools ];
+
+  dependencies = [ coverage ];
```
Comment (superherointj) [inline pkgs/development/python-modules/json2html/default.nix:22]:
```suggestion
```
What happens if you remove this?

File: pkgs/development/python-modules/json2html/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
+  buildPythonPackage,
+  lib,
+  python3,
+  fetchFromGitHub,
+}:
+
+buildPythonPackage rec {
+  pname = "json2html";
+  version = "1.1.1";
```
Comment (superherointj) [inline pkgs/development/python-modules/json2html/default.nix:10]:
```suggestion
  version = "1.3.0";
```
Pypy has 1.3.0.
https://pypi.org/project/json2html/

https://github.com/softvar/json2html/releases/tag/v1.3.0

Why 1.1.1?

File: pkgs/development/python-modules/json2html/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
+  buildPythonPackage,
+  lib,
+  python3,
+  fetchFromGitHub,
```
Comment (superherointj) [inline pkgs/development/python-modules/json2html/default.nix:5]:
```suggestion
  buildPythonPackage,
  fetchFromGitHub,
  lib,
  python3,
```
IMO it's nicer if alphabetical order is kept.

**Comment by superherointj** [comment]
CI errors as:

image

**Comment by superherointj** [comment]
It's missing the python imports check for this package.

**Comment by superherointj** [comment]
There is a problem in fetchFromGitHub. Use owner and repo. And provide the string (not reference).

Also test build of package before pushing code / requesting reviews.

File: pkgs/development/python-modules/json2html/default.nix (hunk 2 above)
Comment (Tochiaha) [reply pkgs/development/python-modules/json2html/default.nix]: now using github 1.3.0

File: pkgs/development/python-modules/json2html/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  build-systems = [ setuptools ];
+
+  # no proper test available
+  doCheck = false;
+
+  pythonImportsCheck = [ "json2html" ];
+
+  meta = {
```
Comment (superherointj) [inline pkgs/development/python-modules/json2html/default.nix:28]: Add changelog:

> https://github.com/softvar/json2html/releases/tag/v${version}

**Comment by superherointj** [comment]
Squash commits.

**Comment by superherointj** [comment]
Errors as:

> ModuleNotFoundError: No module named 'setuptools'

File: pkgs/development/python-modules/json2html/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+  src = fetchFromGitHub {
+    owner = "softvar";
+    repo = "json2html";
+    rev = "v${version}";
+    hash = "sha256-Y+mwJ0p4Q2TKMU8qQvuvo08RiMdsReO7psgXaiW9ntk=";
+  };
+
+  build-systems = [ setuptools ];
+
```
Comment (superherointj) [inline pkgs/development/python-modules/json2html/default.nix:22]:
```suggestion
    nativeBuildInputs = [ setuptools ];
```

**Comment by Tochiaha** [comment]
i believe this "nativeBuildInputs = [ setuptools ];" is not meant to be used, maybe askin @SuperSandro2000 will help.

**Comment by superherointj** [comment]
> i believe this "nativeBuildInputs = [ setuptools ];" is not meant to be used, maybe askin @SuperSandro2000 will help as well.

Did you try to build without it? Build fails. What may not be necessary is the other config.

**Comment by Tochiaha** [comment]
yes. Buiding without it produces the python module required.

**Review by superherointj — APPROVED** [review]

**Pull request merged**